### DISCUSSION TITLE
Beelink Mini5 + Ubuntu 24.04 LTS

### DISCUSSION THREAD

**Discussion by staticgroup** [OpenSurv/OpenSurv#6] [Hardware + Operating system]
I am testing on a beelink mini5 
Business Mini PC, 
Intel 11th Gen N5095 Processor 4 Cores, 
16GB DDR4 
256G SSD, 
4K Dual HDMI (running 1080p dual screens)
Will share results etc as these are VERY cheap on amazon.

**Comment by staticgroup** [comment]
install took 25min , seems to cycle through demo stuff quick will apply rpisurv config now

**Comment by staticgroup** [comment]
9 streams worked well once up. The connecting phase took about twice as long as my pi 4 for 12 streams. 
I tried a second screen with 9 on one screen 6 on the other and it crashed. I will keep trying at it and update my post going forward. I have 30 days to confirm success before I cant return this as I would opt for a dell usff next if this failed. maybe a gen 9 i5, 16gb, 256

**Comment by SvenVD** [comment]
Thanks for testing!

The initial connecting phase indeed takes a little longer as with RpiSurv but this is due to the fact that we need to wait for X windows to report back the window id and this takes a little longer.

Can you describe how and what crashed? Was there something in the logs?

TIP: On slower hardware you can set more streams per screen by [disabling caching](https://github.com/OpenSurv/opensurv/blob/710ffc541626a13d8ce53c9eaca6ce0000d61a4a/surveillance/etc/general.yml#L5). Trade-off will be that cycling to next screen has a loading screen that takes a little longer as the next streams need to start from scratch.

**Reply by staticgroup** [reply]
No screen cycling so will deff do this. The power diff between n100 and this cpu is a few percent. Going to do more testing tommorrow. Also going to try vmwarw with gpu passthrough on my lab . As that's gen 12 i7 so might be very quick.

**Reply by staticgroup** [reply]
There was no helpful logs. Just rtsp conneciviry issues which is neatmtx related . There was a error message at crash i missed will try be quick tomorrow and try various configurations. Note I am using 24.04 server. Would desktop help gpu driver wise ?

**Reply by SvenVD** [reply]
I think it does not matter as long as you have enabled all repositories. Especially restricted since that may contain proprietary drivers for example for nvidia ` add-apt-repository restricted`

**Reply by staticgroup** [reply]
got 15 streams going today no issue, didnt try a second screen again yet, i susepct i had a config issue perhaps. will try second screen tommorrow

**Reply by SvenVD** [reply]
Nice, 15 streams on the Beelink Mini5? Or on the vmware?

**Reply by staticgroup** [reply]
beelink mini, ill do a vmware build in a new post got it up to 16 now. BUT once I go to second monitor even if I drop streams to 6 each monitor it just doesnt load. I have checked formatting seems to be ok.

**Reply by staticgroup** [reply]
2024/11/14 23:45:15 - l_default - ERROR - Stream: screen_manager_1_screen1_stream8: _wait_for_window_to_be_initialized, window not found (14/14. Stopping attempt
2024/11/14 23:45:22 - l_default - ERROR - Stream: screen_manager_1_screen1_stream9: _wait_for_window_to_be_initialized, window not found (14/14. Stopping attempt

**Reply by belyache** [reply]
> Thanks for testing!
> 
> The initial connecting phase indeed takes a little longer as with RpiSurv but this is due to the fact that we need to wait for X windows to report back the window id and this takes a little longer.
> 
> Can you describe how and what crashed? Was there something in the logs?
> 
> TIP: On slower hardware you can set more streams per screen by [disabling caching](https://github.com/OpenSurv/opensurv/blob/710ffc541626a13d8ce53c9eaca6ce0000d61a4a/surveillance/etc/general.yml#L5). Trade-off will be that cycling to next screen has a loading screen that takes a little longer as the next streams need to start from scratch.

Can you tell me, does memory have much effect on how well the system will work? I am looking at the Beelink mini, but through Amazon I am mostly seeing 8GB of RAM compared to the 16GB that StaticGroup was testing with.
I have been using an RPi 3B+ with RPISURV, and could barely support 7 streams. It would be nice if we could get 10 or more.

Thanks,

Glenn

**Reply by SvenVD** [reply]
Rpi is mostly limited by bandwidth of gpu. An x86 machine with a good integrated gpu can display more, here they can display 25 streams with 8G https://github.com/OpenSurv/OpenSurv/discussions/13 . It also depends on codec and resolution of streams etc

**Comment by staticgroup** [comment]
Ok I have it working with 18 streams across two monitors . My indentation was wrong for monitor2.yml 
@SvenVD  I have some useabliltiy feedback is this the right forum or should I write a new post?

**Reply by SvenVD** [reply]
> Ok I have it working with 18 streams across two monitors . My indentation was wrong for monitor2.yml @SvenVD I have some useabliltiy feedback is this the right forum or should I write a new post?

A post in "general" is maybe more suited so we can keep this for the hardware. Thanks for the extensive testing, I think we can promote Beelink Mini5 also as compatible hardware, what do you think?

**Comment by staticgroup** [comment]
I have done further testing and agree, I found the source of my primary issue lets close this out as compatible . Running 24.04 ubuntu server at full patch level on 15/11/24

**Comment by SvenVD** [comment]
I promoted it to compatible device.

**Reply by sborn** [reply]
Is this a Mini S rather than Mini 5?

**Reply by SvenVD** [reply]
It seems like @staticgroup can you confirm?
